Mathlib/Order/Incomparable.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Incomparability relation
+
+Two values in a preorder are incomparable whenever `¬ a ≤ b` and `¬ b ≤ a`.
+
+## Main declarations
+
+* `IncompRel`: The incomparability relation. `IncompRel r a b` means that `a` and `b` are related in
+  neither direction by `r`.
+-/
+
+open Function OrderDual
+
+variable {α : Type*} {a b c d : α}
+
+section Relation
+
+variable (r : α → α → Prop)
+
+/-- The incomparability relation `IncompRel r a b` means `¬ r a b` and `¬ r b a`. -/
+def IncompRel (a b : α) : Prop :=
```

Comment:
What about
```suggestion
def Order.Incomparable (a b : α) : Prop :=
```
? This carries more semantic meaning. I would be happy for `AntisymmRel` to be renamed to `Order.Indistinguishable` if you feel strongly about the names matching